src/error.rs

```diff
@@ -19,6 +21,66 @@ pub enum ProtocolError {
     UnsupportedCompressionFlag(u8),
 }
 
+impl<T> fmt::Display for Error<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Error::Grpc(ref _status, ref _header_map) =>
+                write!(f, "gRPC error"),
+            Error::Protocol(ref _protocol_error) =>
+                write!(f, "Protocol error"),
```

Comment:
A bit of a nit: String descriptions of errors should typically begin with lowercase letters (as per the standard library precedent).

Reply:
Ah, didn't know that. Fixed.

Reply:
I think this is because the `fmt::Display` output of one error is often used as part of the message produced by the `Display` impl for another error, you'll see things like
```rust
enum MyError {
    Io(::std::io::Error),
    Proto(ProtocolError),
}
```rust
enum MyError {
    DiskIo(::std::io::Error),
    NetworkIo(::std::io::Error),
}
impl fmt::Display for MyError {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        match self {
            MyError::DiskIo(ref io_err) => 
                write!(f, "error reading from disk: {}", io_err),
            MyError::NetworkIo(ref io_err) => 
                write!(f, "{} while establishing remote connection", io_err)
        }
    }
}
```
And in this case, the `Display` output for `MyErr` will look nicer if the `Display` impl for `io::Error` produces sentence fragments instead of full sentences.

This is really not terribly important in this case at all, since this isn't a library where we'd expect someone downstream to use our errors as part of their error type, but I thought I'd point it out.
```